src/docfx/lib/git/GitUtility.cs

```diff
@@ -129,6 +129,26 @@ public static string[] GetCommits(string path, string committish, int top = 1)
                     .Split('\n', StringSplitOptions.RemoveEmptyEntries);
         }
 
+        /// <summary>
+        /// Check if remote branch exists
+        /// </summary>
+        public static bool RemoteBranchExists(string remote, string branch)
+        {
+            try
+            {
+                if (GitRemoteProxy != null)
+                {
+                    remote = GitRemoteProxy(remote);
+                }
+
+                return Execute(".", $"ls-remote --heads \"{remote}\" {branch}").Split('\n', StringSplitOptions.RemoveEmptyEntries).Any();
+            }
+            catch
```

Comment:
when is there an exception? can it be implemented without using exception handling

Reply:
when the remote doesn't exists, just return false, I think `the remote doesn't exist` error should be thrown by `restore`